### DISCUSSION TITLE
wrong results in test_commonmatrix

### DISCUSSION THREAD

**Discussion by 1e9abhi1e10** [sympy/sympy#24672] [General]
from https://github.com/sympy/sympy/blob/master/sympy/matrices/tests/test_commonmatrix.py#L354
``` python     
m = Matrix(3, 3, [1, x**2 + 2*x + 1, y, (x + 1)**2, 2, 0, y, 0, 3])
assert m.is_symmetric() # True
assert not m.is_symmetric(simplify=False) #False 
assert m.expand().is_symmetric(simplify=False) #True 
```
But 
``` python
>>> m
Matrix([
[         1, x**2 + 2*x + 1, y],
[(x + 1)**2,              2, 0],
[         y,              0, 3]])

>>> m.transpose()
Matrix([
[             1, (x + 1)**2, y],
[x**2 + 2*x + 1,          2, 0],
[             y,          0, 3]])
```
`m[1,2] = x**2 + 2*x + 1` and `m[2,1] = (x + 1)**2` which are not equal, therefore m is not symmetric and the method would return False.
But current behavior on master is`True`
see also #24671

**Comment by ThePauliPrinciple** [comment]
Surely they are equal in the mathematical meaning?
```python
>>> ((x+1)**2).expand()
x**2 + 2*x + 1
```

**Reply by 1e9abhi1e10** [reply]
``` python
>>> a = (x+1)**2 
>>> b = x**2 + 2*x + 1
>>> a == b
False
```

**Reply by 1e9abhi1e10** [reply]
> ```python
> >>> a = (x+1)**2 
> >>> b = x**2 + 2*x + 1
> >>> a == b
> False
> ```

This is what I got

**Reply by ThePauliPrinciple** [reply]
Sorry, I copy pasted it incorrectly, it's updated now. In general sympy does not check for mathematical equality with `==`, only for structural equality, meaning that they have the exact same node structure.

**Reply by ThePauliPrinciple** [reply]
https://docs.sympy.org/latest/tutorials/intro-tutorial/gotchas.html#equals-signs 

You can read about it here, which uses exactly the example we are discussing here.

**Reply by 1e9abhi1e10** [reply]
Should I compare the results of `sage math` and `sympy`, sorry if not.
In `sage math` I got `False` for this :
``` 
sage: m = Matrix(3, 3, [1, x**2 + 2*x + 1, y, (x + 1)**2, 2, 0, y, 0, 3])
sage: m.is_symmetric()
False
```

**Reply by ThePauliPrinciple** [reply]
I don't know about sage math. But can you please write explicitly whether you agree that in mathematics the following equality holds: `x**2 + 2*x + 1 = (x+1)**2 = (x+1)*(x+1)= x**2 + x*1 + 1*x + 1**2`?

**Reply by 1e9abhi1e10** [reply]
I totally agree with your statement but if we think in this way then what is the use of `expand` and `simplify`

**Reply by ThePauliPrinciple** [reply]
`m.is_symmetric(simplify=True)` is the default in SymPy
https://github.com/sympy/sympy/blob/c4d6b3eb1c7b39832116f7b02baf524a1ffadfba/sympy/matrices/common.py#L1748-L1754

So it checks if it is "mathematically" symmetric if you use `m.is_symmetric()`. Which is what master does for me and what the test is testing for.

If you want to test for "structural" symmetry, you have to set it to `False` (maybe that is the default for sage?)

`expand` will be applied to all elements of a `Matrix`, so both off diagonals will (in this case) expand to the same expression

**Comment by sylee957** [comment]
I consider `simplify` argument for `is_symmetric` design flaw because it can be replaced with `M.applyfunc(simplify).is_symmetric(simplify=False)`

**Reply by 1e9abhi1e10** [reply]
Are these tests look good for you?
``` python
m = Matrix(3, 3, [1, x**2 + 2*x + 1, y, (x + 1)**2, 2, 0, y, 0, 3])
assert m.is_symmetric() # True
assert not m.is_symmetric(simplify=False) #False 
assert m.expand().is_symmetric(simplify=False) #True 
```
